l2discovery.go

```diff
@@ -371,10 +373,17 @@ func getPci(ifaceName string) (aPciAddress exports.PCIAddress, err error) {
 	}
 	stdout = strings.TrimSuffix(stdout, newLineCharacter)
 	spaceIndex := strings.Index(stdout, emptySpaceSeparator)
-	if spaceIndex != -1 {
-		description := stdout[spaceIndex+1:]
+	newLineIndex := strings.Index(stdout, newLineCharacter)
```

Comment:
Why not use regex for this since it is more complex than original parsing:
`func parseLspci(output string) (string, string, error) {
	const regex = `(?m)[^\s]*\s*(.*)$(?m)\s+Subsystem:\s*(.*)$`

	// Compile the regular expression
	re := regexp.MustCompile(regex)

	// Find all matches
	matches := re.FindAllStringSubmatch(output, -1)
	var description, subsystem string
	if len(matches) < 1 {
		return description, subsystem, fmt.Errorf("could not parse lspci output")
	}
	description=matches[0][1]
	subsystem=matches[0][2]
	return description, subsystem, nil
}`
Unit test:
`package main


import "testing"

func Test_parseLspci(t *testing.T) {
	type args struct {
		output string
	}
	tests := []struct {
		name    string
		args    args
		want    string
		want1   string
		wantErr bool
	}{
		{
			name: "ok",
			args: args{
				output: `02:00.0 USB controller: Red Hat, Inc. QEMU XHCI Host Controller (rev 01) (prog-if 30 [XHCI])
        Subsystem: Red Hat, Inc. Device 1100
        Physical Slot: 0-2
        Control: I/O+ Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR+ FastB2B- DisINTx+
        Status: Cap+ 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
        Latency: 0, Cache Line Size: 64 bytes
        Interrupt: pin A routed to IRQ 22
        Region 0: Memory at fe600000 (64-bit, non-prefetchable) [size=16K]
        Capabilities: <access denied>
        Kernel driver in use: xhci_hcd`,
			},
			want: "USB controller: Red Hat, Inc. QEMU XHCI Host Controller (rev 01) (prog-if 30 [XHCI])",
			want1: "Red Hat, Inc. Device 1100",
			wantErr: false,
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got, got1, err := parseLspci(tt.args.output)
			if (err != nil) != tt.wantErr {
				t.Errorf("parseLspci() error = %v, wantErr %v", err, tt.wantErr)
				return
			}
			if got != tt.want {
				t.Errorf("parseLspci() got = %v, want %v", got, tt.want)
			}
			if got1 != tt.want1 {
				t.Errorf("parseLspci() got1 = %v, want %v", got1, tt.want1)
			}
		})
	}
}`

Reply:
thanks this approach is cleaner.. regex isnt my strongest suite.. I have followed your suggestion :)